lib/cron.js

```diff
@@ -590,8 +594,12 @@
 		// and does the real callback logic when it's time.
 
 		function callbackWrapper() {
+			if (self.maxJobCount === self.jobFinishedCount) {
+				self.stop();
+				return;
+			}
+			self.jobFinishedCount++;
```

Comment:
Having this here wont actually work. The reason is that this doesn't always indicate a successful execution. You'd want this line to be closer to the `self.fireOnTick()`.